sync_boostr_data.py

```diff
@@ -0,0 +1,143 @@
+            break
+
+        closed_won_deals = filter(lambda d: d["stage_name"] == "Closed Won", deals)
+        for deal in closed_won_deals:
+            boostr_deal, ok = BoostrDeal.objects.update_or_create(
+                boostr_id=deal["id"],
+                name=deal["name"],
+                advertiser=deal["advertiser_name"],
+                currency=deal["currency"],
+                amount=math.floor(float(deal["budget"])),
+                sales_representatives=','.join(str(d["email"]) for d in deal["deal_members"]),
+                start_date=deal["start_date"],
+                end_date=deal["end_date"],
+            )
+            config.log.debug(f"Upserted deal with boostr_id: {deal['id']}")
+
+            deal_id = deal["id"]
+            deal_products_response = requests.get(f"{config.base_url}/deals/{deal_id}/deal_products", headers=config.headers)
+
+            if deals_response.status_code != 200:
+                raise BoostrApiError(f"Bad response status from /api/deals/{deal_id}/deal_products: {deals_response}")
+
+            deal_products = deal_products_response.json()
+            config.log.debug(f"Fetched {len(deal_products)} deal_products for deal: {deal_id}")
+
+            for deal_product in deal_products:
+                product = BoostrProduct.objects.get(boostr_id = deal_product["product"]["id"])
+                for budget in deal_product["deal_product_budgets"]:
+                    deal_product_budget, ok = BoostrDealProduct.objects.update_or_create(
+                        boostr_deal=boostr_deal,
+                        boostr_product=product,
+                        month=budget["month"],
+                        budget=budget["budget"],
+                    )
+                config.log.debug(f'Upserted {len(deal_product["deal_product_budgets"])} months of budget for product: {product.id} to deal: {deal_id}')
+
+def upsert_products(config: BoostrSyncConfig)-> None:
+    products_params = {
+        "per": 300,
+        "page": 1,
+        "filter": "all",
+    }
+    products_response = requests.get(f"{config.base_url}/products", params=products_params, headers=config.headers)
+    products = products_response.json()
+    if products_response.status_code != 200:
+        raise BoostrApiError(f"Bad response status from /api/products: {products_response}")
+    config.log.info(f"Fetched {(len(products))} products")
+
+    for product in products:
+        BoostrProduct.objects.update_or_create(
+            boostr_id=product["id"],
+            full_name=product["full_name"],
+            campaign_type=get_campaign_type(product["full_name"])
+        )
+    config.log.info(f"Upserted {(len(products))} products")
+
+
+def get_campaign_type(product_full_name: str) -> str:
+    if "CPC" in product_full_name:
+        return "CPC"
+    if "CPM" in product_full_name:
+        return "CPM"
+    return "?"
+
+
+sync_boostr_data()
```

Comment:
Adding this condition will make it easier to unit test functions in this file

```suggestion
if __name__ == "__main__":
    sync_boostr_data()
```